tests/src/Context/FieldContext.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\FieldContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Context\Context;
+
+/**
+ * Context with field management.
+ */
+class FieldContext implements Context {
+
+  /**
+   * The list of field instances before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultFields = array();
+
+  /**
+   * The list of field groups before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultFieldGroups = array();
+
+  /**
+   * Remember the list of fields existing by default in the site.
+   *
+   * @BeforeScenario
+   */
+  public function rememberCurrentFields() {
+    $this->defaultFields = field_info_field_map();
+  }
+
+  /**
+   * Removes any fields created by a scenario.
+   *
+   * @AfterScenario
+   */
+  public function resetFields() {
+    if (empty($this->defaultFields)) {
+      return;
+    }
+
+    $current_fields = field_info_field_map();
+    $purge_field_schema = FALSE;
+    foreach ($current_fields as $field_name => $field_info) {
+      if (!isset($this->defaultFields[$field_name])) {
+        $field_info = field_info_field($field_name);
+        if (!is_null($field_info)
+          && !$field_info['locked']
+          && !$field_info['deleted']) {
+          field_delete_field($field_name);
+          $purge_field_schema = TRUE;
+        }
+      }
+      else {
+        // We check if a test has not created an instance of an existing field.
```

Comment:
This part of the code is very interesting and very well implemented.
Unfortunately, I think, it's out of the scope of this ticket because I don't see any case on features with the case "create a instance between an existing field and an entity type"
But I think this logic to check field's changes, should be implemented on a follow up ticket. Furthermore, I think this code could be improved with other checks (for example: other values into the Field object like "type", "translatable", "locked", ... )